src/platform_impl/ios/uikit/gesture_recognizer.rs

```diff
@@ -0,0 +1,141 @@
+use crate::platform_impl::platform::view::WinitView;
+use icrate::Foundation::{CGFloat, NSInteger, NSObject, NSUInteger};
+use objc2::encode::{Encode, Encoding};
+use objc2::rc::Id;
+use objc2::runtime::Sel;
+use objc2::{extern_class, extern_methods, msg_send_id, mutability, ClassType};
+
+// https://developer.apple.com/documentation/uikit/uigesturerecognizer
+extern_class!(
+    #[derive(Debug, PartialEq, Eq, Hash)]
+    pub(crate) struct UIGestureRecognizer;
+
+    unsafe impl ClassType for UIGestureRecognizer {
+        type Super = NSObject;
+        type Mutability = mutability::InteriorMutable;
+    }
+);
+
+extern_methods!(
+    unsafe impl UIGestureRecognizer {
+        #[method(state)]
+        pub fn state(&self) -> UIGestureRecognizerState;
+
+        #[method(initWithTarget:action:)]
+        pub fn initWithTarget(&self, target: WinitView, action: Sel);
+    }
+);
+
+unsafe impl Encode for UIGestureRecognizer {
+    const ENCODING: Encoding = Encoding::Object;
+}
+
+// https://developer.apple.com/documentation/uikit/uigesturerecognizer/state
+#[derive(Debug)]
+#[allow(dead_code)]
+#[repr(isize)]
+pub enum UIGestureRecognizerState {
```

Comment:
Nit: Use a newtype that wraps `NSInteger`, similar to how it's done with `UIScreenOverscanCompensation` in `uikit/screen.rs`